pkg/cmd/compile/options_test.go

```diff
@@ -90,10 +101,12 @@ func TestCompileOptions_Run(t *testing.T) {
 	}()
 
 	t.Run("no style is true", func(t *testing.T) {
-		defer monkey.UnpatchAll()
-		mockDetectProjectAndStack()
-		mockGenerateSpec()
-		mockWriteFile()
+		m1 := mockDetectProjectAndStack()
+		m2 := mockGenerateSpec()
+		m3 := mockWriteFile()
+		defer m1.UnPatch()
+		defer m2.UnPatch()
+		defer m3.UnPatch()
```

Comment:
> You can choose to use convey, then there is no need to write unpatch :)

Good idea, but this would increase the cost of understanding, and a more in-depth discussion should be had.